examples/parallel_memmap.py

```diff
@@ -1,19 +1,95 @@
-import shutil
-import os
+
+##############################################################################
+# Speed-up processing with large data array
+##############################################################################
+# 
+# We create a large data array from which an average is computed for several
+# slices.
+
 import numpy as np
 
+data = np.random.random((int(1e7),))
+window_size = int(5e5)
+slices = [slice(start, start + window_size)
+          for start in range(0, data.size - window_size, int(1e5))]
+
+###############################################################################
+# The ``slow_mean`` function introduce a :func:`time.sleep` call to simulate a
+# more expensive computation cost for which parallel computing is beneficial.
+# Otherwise, for very fast sequential processing, parallel processing is not
+# adapted due some extra overheads (workers creations, communication, etc.).
+
+import time
+
+
+def slow_mean(data, sl):
+    """Simulate a larger processing"""
+    time.sleep(0.005)
+    return data[sl].mean()
+
+
+###############################################################################
+# First, we will evaluate the sequential computing on our problem.
+
+tic = time.time()
+results = [slow_mean(data, sl) for sl in slices]
+toc = time.time()
+print('\nElapsed time computing the average on couple of slices {:.2f} s'
+      .format(toc - tic))
+
+###############################################################################
+# :class:`joblib.Parallel` is used to compute in parallel the average on each
+# slices using 2 workers.
+
 from joblib import Parallel, delayed
-from joblib import load, dump
+
+
+tic = time.time()
+results = Parallel(n_jobs=2)(delayed(slow_mean)(data, sl) for sl in slices)
+toc = time.time()
+print('\nElapsed time computing the average on couple of slices {:.2f} s'
+      .format(toc - tic))
+
+###############################################################################
+# Surprisingly (sic) the parallel processing is slower than the sequential
+# processing. Indeed, ``data`` is hashed at each call of ``slow_mean``, leading
+# to an important time overhead. A solution is to dump this array to a memmap
+# and pass the memmap to :class:`joblib.Parallel`.
+
+from joblib import dump, load
+
+filename_memmap = 'data.pkl'
+dump(data, filename_memmap)
+data = load(filename_memmap, mmap_mode='r')
+
+tic = time.time()
+results = Parallel(n_jobs=2)(delayed(slow_mean)(data, sl) for sl in slices)
+toc = time.time()
+print('\nElapsed time computing the average on couple of slices {:.2f} s\n'
+      .format(toc - tic))
+
+###############################################################################
+# Therefore, dumping large ``data`` array ahead of calling
+# :class:`joblib.Parallel` can speed-up significantly the processing by
+# removing some overheads.
+
+###############################################################################
+# Output write access within :class:`joblib.Parallel`
```

Comment:
I would use as a title "Memmap for shared memory", and explain that it works only for numpy tables.

Reply:
Or "Writable memmap for shared memory" (because before the memmap was already 'shared' for reading it?)